src/targets/gpu/include/migraphx/gpu/gemm_impl.hpp

```diff
@@ -24,28 +24,492 @@
+        // A and args[0] as B in calling the rocblas_gemm.
+        return pack(ctx.get_stream().get_rocblas(),
+                    transb ? rocblas_operation_transpose : rocblas_operation_none,
+                    transa ? rocblas_operation_transpose : rocblas_operation_none,
+                    n,
+                    m,
+                    k,
+                    alpha_v,
+                    args[1].data(),
+                    arg_type,
+                    ldb,
+                    b_stride,
+                    args[0].data(),
+                    arg_type,
+                    lda,
+                    a_stride,
+                    beta_v,
+                    args[2].data(),
+                    output_type,
+                    ldc,
+                    c_stride,
+                    is_3inputs ? args[3].data() : args[2].data(),
+                    output_type,
+                    ldd,
+                    d_stride,
+                    num_matrices,
+                    compute_type,
+                    rocblas_gemm_algo_solution_index,
+                    solution_idx,
+                    flag);
+    }
+
+    void run(context& ctx, const std::vector<argument>& input_args, int32_t solution_idx = 0) const
+    {
+        if(strided_batched)
+        {
+            auto gemm_args =
+                create_strided_batched_gemm_args(ctx, input_args, int8_flag, solution_idx);
+            rocblas_invoke(&rocblas_gemm_strided_batched_ex, gemm_args);
+        }
+        else
+        {
+            auto gemm_args = create_gemm_args(ctx, input_args, int8_flag, solution_idx);
+            rocblas_invoke(&rocblas_gemm_ex, gemm_args);
+        }
+    }
+
+    auto validate(context& ctx, const std::vector<shape>& input_shapes, int32_t solution_idx) const
+    {
+        std::vector<argument> input_args;
+        std::transform(input_shapes.begin(),
+                       input_shapes.end(),
+                       std::back_inserter(input_args),
+                       [](const shape& x) { return to_gpu(generate_argument(x)); });
+
+        return validate(ctx, input_args, solution_idx);
+    }
+
+    /**
+     * Checks a particular solution for validity by running it with the flag
+     * rocblas_gemm_flags_check_solution_index, and returns the result
+     */
+    auto validate(context& ctx, const std::vector<argument>& input_args, int32_t solution_idx) const
+    {
+        if(strided_batched)
+        {
+            auto gemm_args = create_strided_batched_gemm_args(
+                ctx, input_args, rocblas_gemm_flags_check_solution_index, solution_idx);
+            return rocblas_invoke(&rocblas_gemm_strided_batched_ex, gemm_args);
+        }
+        else
+        {
+            auto gemm_args = create_gemm_args(
+                ctx, input_args, rocblas_gemm_flags_check_solution_index, solution_idx);
+            return rocblas_invoke(&rocblas_gemm_ex, gemm_args);
+        }
+    }
+
+    void print_args() const
```

Comment:
This is mostly for debugging. You can also check if rocBLAS has any environment flags that would dump args for `rocBLAS-bench`. In that  case, we wouldn't need this. Or we can make `print_args()` to print args such that we can directly pass to `rocBLAS-bench`. 
This is just an improvement, You can work on it if you wish after everything else is fixed/done.

Reply:
There **are** environment flags like that; see [API_Reference_Guide logging-in-rocblas](https://rocblas.readthedocs.io/en/rocm-5.2.3/API_Reference_Guide.html?highlight=rocBLAS-bench#logging-in-rocblas).  I think that takes care of this with no action required in code.